From 6b4a63b8ff3fc1e3069d8ca9dc04837866b3b280 Mon Sep 17 00:00:00 2001
From: Nikita Yushchenko <nikita.yoush@cogentembedded.com>
Date: Sat, 3 Aug 2024 02:01:39 +0500
Subject: [PATCH 142/157] net: ethernet: renesas: rswitch: introduce
 rswitch_lib

This library module will contain ring management and other operations
that does not require rswitch register access.

The library is intended both by rswitch module and by modules
implementing virtual ports.

Signed-off-by: Nikita Yushchenko <nikita.yoush@cogentembedded.com>

diff --git a/drivers/net/ethernet/renesas/Kconfig b/drivers/net/ethernet/renesas/Kconfig
index 04acc0333c44..4f8e4c875790 100644
--- a/drivers/net/ethernet/renesas/Kconfig
+++ b/drivers/net/ethernet/renesas/Kconfig
@@ -46,9 +46,13 @@ config RTSN_PTP
 	tristate
 	default n
 
+config RENESAS_ETHER_SWITCH_LIB
+	tristate
+
 config RENESAS_ETHER_SWITCH
 	tristate "Renesas Ethernet Switch support"
 	depends on ARCH_RENESAS || COMPILE_TEST
+	select RENESAS_ETHER_SWITCH_LIB
 	select CRC32
 	select MII
 	select PHYLIB
diff --git a/drivers/net/ethernet/renesas/Makefile b/drivers/net/ethernet/renesas/Makefile
index 1848c370a555..e12589f402e6 100644
--- a/drivers/net/ethernet/renesas/Makefile
+++ b/drivers/net/ethernet/renesas/Makefile
@@ -11,4 +11,5 @@ obj-$(CONFIG_RAVB) += ravb.o
 
 obj-$(CONFIG_RTSN_PTP) += rtsn_ptp.o
 obj-$(CONFIG_RTSN) += rtsn.o
+obj-$(CONFIG_RENESAS_ETHER_SWITCH_LIB) += rswitch_lib.o
 obj-$(CONFIG_RENESAS_ETHER_SWITCH) += rswitch.o
diff --git a/drivers/net/ethernet/renesas/rswitch_lib.c b/drivers/net/ethernet/renesas/rswitch_lib.c
new file mode 100644
index 000000000000..b7e4c24ce02a
--- /dev/null
+++ b/drivers/net/ethernet/renesas/rswitch_lib.c
@@ -0,0 +1,6 @@
+// SPDX-License-Identifier: GPL-2.0
+
+#include <linux/module.h>
+#include "rswitch_lib.h"
+
+MODULE_LICENSE("GPL v2");
diff --git a/drivers/net/ethernet/renesas/rswitch_lib.h b/drivers/net/ethernet/renesas/rswitch_lib.h
new file mode 100644
index 000000000000..2978ba06bf2a
--- /dev/null
+++ b/drivers/net/ethernet/renesas/rswitch_lib.h
@@ -0,0 +1,6 @@
+// SPDX-License-Identifier: GPL-2.0
+
+#ifndef RSWITCH_LIB_H
+#define RSWITCH_LIB_H
+
+#endif
-- 
2.25.1

